mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/handlers/JobArtifactRouteHandlerImpl.java

```diff
@@ -76,4 +77,17 @@ public CompletionStage<JobArtifactProto.UpsertJobArtifactResponse> upsert(JobArt
             }
         });
     }
+
+    @Override
+    public CompletionStage<BaseResponse> upsertJobArtifactToCache(JobArtifactProto.JobArtifactToCacheRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                mantisStorageProvider.addNewJobArtifactToCache(request.getClusterID(), request.getArtifactID());
```

Comment:
Do we need to send a message to the resource cluster actor instead and have it deal with the persistence, etc..?